reader-sdk-react-native-quickstart/android/app/src/main/java/com/rnreadersdksample/MainApplication.kt

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 Square Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.rnreadersdksample
+
+import android.app.Application
+import com.facebook.react.ReactApplication
+import com.facebook.react.ReactNativeHost
+import com.facebook.react.ReactPackage
+import com.facebook.react.PackageList
+import com.facebook.soloader.SoLoader
+import com.squareup.sdk.reader.ReaderSdk
+import com.rnreadersdksample.MainApplication
+import android.content.Context
+import com.facebook.react.ReactInstanceManager
+import java.lang.ClassNotFoundException
+import java.lang.NoSuchMethodException
+import java.lang.IllegalAccessException
+import java.lang.reflect.InvocationTargetException
+
+class MainApplication : Application(), ReactApplication {
+    private val mReactNativeHost: ReactNativeHost = object : ReactNativeHost(this) {
+        override fun getUseDeveloperSupport(): Boolean {
+            return BuildConfig.DEBUG
+        }
+
+        override fun getPackages(): List<ReactPackage> {
+            // Packages that cannot be autolinked yet can be added manually here, for example:
+            // packages.add(new MyReactNativePackage());
+            return PackageList(this).packages
+        }
+
+        override fun getJSMainModuleName(): String {
+            return "index"
+        }
+    }
+
+    override fun getReactNativeHost(): ReactNativeHost {
+        return mReactNativeHost
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        SoLoader.init(this,  /* native exopackage */false)
+        ReaderSdk.initialize(this)
+        initializeFlipper(this, reactNativeHost.reactInstanceManager)
+    }
+
+    companion object {
+        /**
+         * Loads Flipper in React Native templates. Call this in the onCreate method with something like
+         * initializeFlipper(this, getReactNativeHost().getReactInstanceManager());
+         *
+         * @param context
+         * @param reactInstanceManager
+         */
+        private fun initializeFlipper(
+                context: Context, reactInstanceManager: ReactInstanceManager) {
+            if (BuildConfig.DEBUG) {
+                try {
+                    /*
```

Comment:
nit: please format the comment with the correct indents.

Reply:
done

Reply:
done